oeps/architectural-decisions/oep-XXXX-micro-frontend-domains.rst

```diff
@@ -0,0 +1,208 @@
+
+It is thus a good time to consider these practices - specified below - weighing both the development cost of implementation and future recurring ones against their benefits.
+
+
+*************
+Specification
+*************
+
+We propose a compromise between the monolithic codebase and full MFE independence based on Domain Driven Development: the Open edX frontend will be split into formal top-level domains, including but not limited to LMS and CMS, each of which will be implemented as a Container SPA.  The container, in turn, will render multiple partly-independent MFEs as sub-components.  The latter will be re-engineered to conform to the hosting domain's interfaces.
+
+This is to be a full re-platforming, to be undertaken as a single long-term project prior to release.  It is in opposition to the currently in-progress gradual conversion of single pages to fully independent MFEs that can be linked to from the pre-existing monolithic framework; the opposite, injecting legacy pages into a domain SPA as components, would be unjustifiable.  The difficulty of working with the legacy codebase is the main reason why it was decided to move to MFEs in the first place!
+
+On the other hand, the current gradual conversion to fully independent MFEs need not be interrupted; the domain-based re-platforming proposed here can be executed in parallel.  Why?  Converting an already fully independent MFE to a domain-hosted one is much less costly than the initial corresponding extraction from the monolith.  And conversely, it is conceivable that a particular monolithic page, after extraction directly into a domain-hosted MFE, could then be cheaply backported *the other way*, into a fully-independent one for earlier release while the re-platforming project is incomplete.
+
+As this OEP is developed, we will further investigate:
+
+#. Integration strategy
+#. Domain SPAs and their interfaces
+#. Shared dependencies
+#. MFEs as components
+
+Discovery
+---------
+
+* Are there any further top-level architectural elements to consider?
+* What is the relationship of this project to the Modular Learning effort?
+
+  * Both are long-term projects, but beyond that, is there anything else that could be coordinated between them?
+  * Should this project target the modular codebase exclusively, ignoring ``edx-platform``?  (Relevant if the APIs are incompatible between them.)
+
+Integration strategy
+====================
+
+To be defined.
+
+Discovery
+---------
+
+At time of writing these appear to commonly recommended strategies for integration of MFEs into modular container apps:
+
+* `Run-time integration via JavaScript <https://martinfowler.com/articles/micro-frontends.html#Run-timeIntegrationViaJavascript>`_
+* `Run-time integration via Web Components <https://martinfowler.com/articles/micro-frontends.html#Run-timeIntegrationViaWebComponents>`_
```

Comment:
in the FWG meeting today the question of "what's going to be around in 5-10 years?" was raised in the context of which integration strategy to use. my gut feeling is web components would be a good choice based on that criteria

Reply:
FWIW, there is some desire/interest to try to have a web components implementation available for the components in the Paragon design system so that they could be potentially used in non-React environments (definitely an open question in what that would look like practically, but one of our long-term "vision" goals for Paragon is for it to become more framework agnostic in its usage).